### ISSUE TITLE
L039 interfering with L003 in SQLFluff fix

### ISSUE THREAD

**Issue by gladysteh99** [sqlfluff/sqlfluff#3705]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

When fixing the indentation, L039 doesn't recognize that it is an indent and proceed to fix the unnecessary white space (reverting the fix by L003)

### Expected Behaviour

Only insert the indents as detected by L003. Expected behavior is achieved by calling 
`sqlfluff fix testfile.sql --dialect snowflake --rules L003`

### Observed Behaviour

L039 removed the "indents" as it recognized it as unnecessary whitespace.

### How to reproduce

`sqlfluff fix testfile.sql --dialect snowflake`
[testfile.sql.zip](https://github.com/sqlfluff/sqlfluff/files/9238081/testfile.sql.zip)

### Dialect

snowflake

### Version

1.2.1

### Configuration

```[sqlfluff]
exclude_rules = L016, L031, L034, L035, L051, L061, L044

[sqlfluff:indentation]
indented_using_on = False
indented_on_contents = True

[sqlfluff:rules:L003]
hanging_indents = False
indent_unit = tab
tab_space_size = 4

[sqlfluff:rules:L004]
indent_unit = tab
tab_space_size = 4

[sqlfluff:rules:L010]
capitalisation_policy = upper

[sqlfluff:rules:L019]
comma_style = leading

# [sqlfluff:rules:L028]
# force_enable = True
# single_table_references = qualified

[sqlfluff:rules:L030]
#Keywords
extended_capitalisation_policy = upper

[sqlfluff:rules:L052]
multiline_newline = True
require_final_semicolon = True
```

### Are you willing to work on and submit a PR to address the issue?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by gladysteh99** [comment]
[testfile.txt](https://github.com/sqlfluff/sqlfluff/files/9242436/testfile.txt)
For some reason github didn't accept .sql format so I uploaded a .zip version in the "How to Reproduce" section but here is the .txt formatted one.

**Comment by alanmcruickshank** [comment]
I'm going to loop this into the overall whitespace issue. Dependencies between how we recreate whitespace in places is a bit convoluted. This should be much easier to address once that's solved.

**Comment by alanmcruickshank** [comment]
I believe this is resolved by #3824. Based on my testing I think it's sorted, it certainly resolves the issue with the `LATERAL FLATTEN` on line 19. Given we're likely to merge that PR regardless, I suggest we merge and then re-assess.

**Comment by alanmcruickshank** [comment]
I'm going to mark this as solved now. I just tested that file and it now indents properly since merging #3824.
